src/dmd/dtemplate.d

```diff
@@ -7287,7 +7287,20 @@ extern (C++) class TemplateInstance : ScopeDsymbol
                 }
                 TemplateInstance ti = sa.isTemplateInstance();
                 Declaration d = sa.isDeclaration();
-                if ((td && td.literal) || (ti && ti.enclosing) || (d && !d.isDataseg() && !(d.storage_class & STC.manifest) && (!d.isFuncDeclaration() || d.isFuncDeclaration().isNested()) && !isTemplateMixin()))
+                /* Have nested arguments when the `alias` parameter...
+                 * 1. `td` is a template function literal
+                 * 2. `ti` is a nested template instantiation
+                 *    (See https://issues.dlang.org/show_bug.cgi?id=9578)
+                 * 3. `d` is a local variable or nested function
+                 * 4. `td` is a nested template function
+                 *    (See https://issues.dlang.org/show_bug.cgi?id=15298)
+                 */
+                if ((td && td.literal) ||
```

Comment:
Is there anyway this if statement could be naturally broken up, it's a bit unwieldy. I can read it but people seem to enjoy bolting new leafs into their if expressions so maybe best to break this one up early.

Reply:
Convert the body into a nested function, and then break up each condition into:
```
if (a)
    fun();
else if (b)
   fun();
...
```